backend/app/dependencies.py

```
@@ -79,10 +79,10 @@ def require_admin(current_user: User = Depends(get_current_user)) -> User:
 
 def require_professor(current_user: User = Depends(get_current_user)) -> User:
```

Ruff (0.15.7)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)